### ISSUE TITLE
jax.random.PRNGKey(seed) -> jax.random.key(seed)

### ISSUE THREAD

**Issue by ealt** [Astera-org/simplexity#31]
Let's switch to `jax.random.key(seed)` instead of `jax.random.PRNGKey(seed)`, see:

```python
PRNGKey(seed: 'int | ArrayLike', *, impl: 'PRNGSpecDesc | None' = None) -> 'Array'
    Create a legacy PRNG key given an integer seed.
    
    This function produces old-style legacy PRNG keys, which are arrays
    of dtype ``uint32``. For more, see the note in the `PRNG keys
    <https://jax.readthedocs.io/en/latest/jax.random.html#prng-keys>`_
    section. When possible, :func:`jax.random.key` is recommended for
    use instead.
```

_Originally posted by @hrbigelow in https://github.com/Astera-org/simplexity/pull/29#discussion_r2029709442_
